src/storage-client/src/sink.rs

```diff
@@ -287,23 +296,43 @@ pub async fn build_kafka(
             .await?;
 
         let progress_client = Arc::new(progress_client);
-        let latest_ts = determine_latest_progress_record(
```

Comment:
Good catch, yes! I became aware of this too during the discussions in the other threads. I will definitely reorder this so that it happens after `init_transactions` once Sean's fixes are in and I rebase on top of them